include/rosdiscover-clang/ApiCall/Calls/PublishCall.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include "../RosApiCall.h"
+#include "./Util.h"
+
+namespace rosdiscover {
+namespace api_call {
+
+class PublishCall : public BareRosApiCall {
+public:
+  using BareRosApiCall::BareRosApiCall;
+
+  RosApiCallKind const getKind() const override {
+    return RosApiCallKind::PublishCall;
+  }
+
+  clang::Expr const * getNameExpr() const override {
+    return getCallExpr();
+  }
+
+  const std::string getPublisher() const {
+    llvm::outs() << "DEBUG [PublishCall] Publish call is : ";
+    getCallExpr()->dump();
+    llvm::outs() << "\n";
+
+    const auto *memberCall = clang::dyn_cast<clang::CXXMemberCallExpr>(getCallExpr());
+    if (memberCall == nullptr) {
+      llvm::outs() << "ERROR [PublishCall] Publish call is not a CXXMemberCallExpr: ";
+      getCallExpr()->dump();
+      llvm::outs() << "\n";
+      return nullptr;
+    }
+    
+    const clang::ValueDecl *decl = getCallerDecl("PublishCall", memberCall);
+    if (decl == nullptr) {
+      llvm::outs() << "ERROR [PublishCall] Decl is null: ";
+      memberCall->dump();
+      llvm::outs() << "\n";
+      return nullptr;
+    }
+
+    llvm::outs() << "DEBUG [PublishCall] decl: ";
+    decl->dump();
+    llvm::outs() << "\n";
+
+    const auto *identifier = decl->getIdentifier();
+    if (identifier == nullptr) {
+      llvm::outs() << "ERROR [PublishCall] Decl identifier is null: ";
+      decl->dump();
+      llvm::outs() << "\n";
+      return nullptr;
+    }
+    llvm::outs() << "DEBUG [PublishCall] Callee is: " << identifier->getName() << "\n";
+
+    return identifier->getName().str();
```

Comment:
You may want to normalize this name before you return it, but for the sake of this PR, it's fine. Maybe you can do that in a subsequent enhancement PR?

Reply:
The function from liftoff doesn't work here because it assumes it's a valueDecl, which is a stronger assumption than what I can make here. So I'd just keep this for now until we have symbolic publishers